src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/host-form/host-form.component.html

```diff
@@ -1,91 +1,95 @@
-<div class="cd-col-form"
-     *cdFormLoading="loading">
-  <form name="hostForm"
-        #formDir="ngForm"
-        [formGroup]="hostForm"
-        novalidate>
-    <div class="card">
-      <div i18n="form title|Example: Create Pool@@formTitle"
-           class="card-header">{{ action | titlecase }} {{ resource | upperFirst }}</div>
+<cd-modal [pageURL]="pageURL"
```

Comment:
I'm not happy with this approach, because it changes the behavior of the component. According to the naming this is a form. But with your changes it is a modal dialog that embeds a form.

To make this behavior clear, change the name of the component to `cd-host-modal-form` or add a new component that wraps the existing `cd-host-form` component. IMO the latter is the better approach.

Reply:
@votdev I am not changing the component. My intention was to change how that component is visually represented. Earlier, it was showing in a card like view, but right now its in a modal form. So I assumed this is the best way to go for it. 

> By the way, why it is necessary to change the established Ceph Dashboard workflow to show forms in pages to modal dialogs?

I proposed this to make the form reusable in other areas, like for eg. in the Cluster Expansion workflow. Earlier it redirects the form to a new area which makes it looks detached with the host list. But for the forms I think its much more convenient to show what's in the background and where does the user accessed the form. So the proposition was accepted on a weekly standup and I went with this. Eventually my plan is to apply this all over the dashboard forms once this is merged in master. https://tracker.ceph.com/issues/51516